examples/rules.dl

```diff
@@ -0,0 +1,48 @@
+// Is this symbol an access path to some data?
+.decl isAccessPath(x: symbol)
+// Is this symbol a label.
+.decl isLabel(x: symbol)
+// A data flow edge
+.decl edge(src: symbol, tgt: symbol)
+ // A data flow path
+.decl path(src: symbol, tgt: symbol)
+// Definitely has this label.
+.decl hasLabel(v: symbol, label: symbol)
+// Definitely has this label.
+.decl claimHasLabel(v: symbol, label: symbol)
+// Definitely does not have this label.
+// This is dual of `hasLabel`. We need a better name.
+.decl mayNotHaveLabel(v: symbol, label: symbol)
+// May have the label
+.decl mayHaveLabel(v: symbol, label: symbol)
+// Label l is preserved along the edge.
+.decl preservesClaim(src: symbol, tgt: symbol, label: symbol)
+
+// Some inference rules
+
+// Symbols used in edges are access paths
+isAccessPath(x) :- edge(x, _). 
+isAccessPath(y) :- edge(_, y). 
+
+// Symbols used in hasLabel or mayHaveLabel are labels
+isLabel(l) :- claimHasLabel(_, l). 
```

Comment:
Bit of a nit, but can we expand "l" to "label" throughout? Mistook this for the number '1' and was somewhat confused for a second.

Reply:
Of course. I am about to send  a separate  PR that defines the base relations that we need for data flow analysis. I will keep this in mind.

Reply:
If we want to make this to map to IFC concepts, it might be useful to reserve the term "Label" for the sets and give another name to the elements of the sets.

Reply:
As another option: I sometimes use "ell" as a reference to LaTeX's `\ell` for label variables. This way it's also distinct from the name of the type.